### PR TITLE
MOTOR-597 Stop using deprecated loop arguments and other improvements

### PR DESCRIPTION
**Pull request by ShaneHarvey**
Jira: https://jira.mongodb.org/browse/MOTOR-597

### CODE REVIEW

File: motor/frameworks/asyncio/__init__.py [hunk 1]
```diff
@@ -97,12 +97,12 @@ def copy(_future):
         else:
             chained.set_result(return_value)
 
-    future._future.add_done_callback(functools.partial(loop.add_callback, copy))
```
Comment (prashantmital) [inline motor/frameworks/asyncio/__init__.py:100]: I had been a bit confused by this back when I was working on the PR to drop older Pythons. can you clarify what is `future._future` and why we don't need it anymore?

Comment (ShaneHarvey) [reply motor/frameworks/asyncio/__init__.py:100]: This was and still is totally broken and completely untested. I'll open a new ticket for it.

Comment (prashantmital) [reply motor/frameworks/asyncio/__init__.py:100]: IIRC `to_list` breaks without this method. I tried to remove it when I was updating the framework files but couldn't figure out a way to get rid of the damn thing.

Comment (ShaneHarvey) [reply motor/frameworks/asyncio/__init__.py]: I'll fix this in https://jira.mongodb.org/browse/MOTOR-600.

**Pull request merged**